src/main/kotlin/org/openrewrite/RecipeMarkdownGenerator.kt

```diff
@@ -1049,8 +1049,22 @@ import TabItem from '@theme/TabItem';
                 newLine()
             }
 
-            writeSourceLinks(recipeDescriptor, origin)
+            if (getLicense(origin) == License.Proprietary) {
+                //language=markdown
+                writeln(
+                    """
+                    ## Recipe source
+
+                    This recipe is only available to users of [Moderne](https://docs.moderne.io/).
+
+                    """.trimIndent()
+                )
+            } else {
+                writeSourceLinks(recipeDescriptor, origin)
+            }
```

Comment:
I'd perhaps push down this conditional into `writeSourceLinks`, just to keep the code readable here.